src/components/constructor-form/constructor-form.js

```diff
@@ -0,0 +1,384 @@
+      image: editableConstructorElement.image || '',
+      uaName: editableConstructorElement.name[0].value || '',
+      enName: editableConstructorElement.name[1].value || '',
+      material: editableConstructorElement.material._id || '',
+      color: editableConstructorElement.color._id || '',
+      available: editableConstructorElement.available || false,
+      default: editableConstructorElement.default || false,
+      basePrice: +editableConstructorElement.basePrice[1].value / 100 || 0
+    },
+
+    onSubmit: () => {
+      const constructorElement = createConstructor(values);
+      history.goBack()
+      if(isEdit){
+        return dispatch(constructorElementMethod({
+          constructorElement,
+          id:editableConstructorElement._id
+        }))
+      }
+      return dispatch(constructorElementMethod({
+        constructorElement,
+        id:model._id
+      }))
+    }
+  });
+
+  const handleMaterial = (e) => {
+    setFieldValue('material', e.target.value);
+    setMaterialColors(list.filter(el=>el._id===e.target.value)[0].colors);
+  }
+  const handleMaterialColor = (e) => {
+    setFieldValue('color', e.target.value);
+  }
+
+  const checkboxes =(checkBoxName, label)=>[
+    {
+      id: `${checkBoxName}`,
+      dataCy: `${checkBoxName}`,
+      value: values[`${checkBoxName}`],
+      checked: values[`${checkBoxName}`],
+      color: 'primary',
+      label,
+      handler: () => setFieldValue(`${checkBoxName}`, !values[`${checkBoxName}`])
+    }
+  ];
+
+  const handleImageLoad = (e, callback) => {
+    if (e.target.files && e.target.files[0]) {
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        callback(event);
+      };
+      reader.readAsDataURL(e.target.files[0]);
+    }
+  };
+
+  const handleLoadConstructorImage = (e) => {
+    handleImageLoad(e, (event) => {
+      setFieldValue('image', event.target.result);
+      setConstructorImg(event.target.result);
+    });
+  };
+
+  const textField = (inputValue, inputName, inputLabel, land) => (<TextField
+    data-cy={land?`${land}${inputName}`:`${inputName}`}
+    id={land?`${land}${inputName}`:`${inputName}`}
+    className={styles.textField}
+    variant='outlined'
+    label={inputLabel}
+    value={inputValue}
+    onChange={handleChange}
+    error={touched.inputName && !!errors.inputName}
+  />)
+
+  const selectField = (selectValue,
+    selectChangeAction,
+    selectItemsList,
+    inputLabel,
+    defaultValue) => (<FormControl className={styles.formControl}>
+    <InputLabel id={`multiple-${selectValue}-label`}>
+      {inputLabel}
+    </InputLabel>
+    <Select
+      variant='outlined'
+      labelId={`multiple-${selectValue}-label`}
+      id={`multiple-${selectValue}`}
+      onChange={selectChangeAction}
+      input={<Input />}
+      value={defaultValue || ''}
+      MenuProps={MenuProps}
+      disabled={!selectItemsList.length}
+    >
+      {selectItemsList?selectItemsList.map((selectItem) => (
```

Comment:
We have lodash package for it in our project. https://lodash.com/docs/4.17.15#map

Reply:
fixed